src/lib/sb-file-uploader-hoc.jsx

```diff
@@ -0,0 +1,241 @@
+                    this.resetFileInput();
+                }
+                this.props.closeFileMenu();
+            }
+        }
+        // called when file upload raw data is available in the reader
+        onload () {
+            if (this.reader) {
+                this.props.onLoadingStarted();
+                const filename = this.fileToUpload && this.fileToUpload.name;
+                this.props.vm.loadProject(this.reader.result)
+                    .then(() => {
+                        this.props.onLoadingFinished(this.props.loadingState, true);
+                        // Reset the file input after project is loaded
+                        // This is necessary in case the user wants to reload a project
+                        if (filename) {
+                            const uploadedProjectTitle = this.getProjectTitleFromFilename(filename);
+                            this.props.onUpdateProjectTitle(uploadedProjectTitle);
+                        }
+                        this.resetFileInput();
+                    })
+                    .catch(error => {
+                        log.warn(error);
+                        this.props.intl.formatMessage(messages.loadError); // eslint-disable-line no-alert
+                        this.props.onLoadingFinished(this.props.loadingState, false);
+                        // Reset the file input after project is loaded
+                        // This is necessary in case the user wants to reload a project
+                        this.resetFileInput();
+                    });
+            }
+        }
+        handleStartSelectingFileUpload () {
+            // open filesystem browsing window
+            this.fileInput.click();
+        }
+        setFileInput (input) {
+            this.fileInput = input;
+        }
+        render () {
+            const fileInput = (
+                <input
+                    accept=".sb,.sb2,.sb3"
+                    ref={this.setFileInput}
+                    style={{display: 'none'}}
+                    type="file"
+                    onChange={this.handleChange}
+                />
+            );
+            const {
+                /* eslint-disable no-unused-vars */
+                closeFileMenu: closeFileMenuProp,
+                isLoadingUpload,
+                isShowingWithoutId,
+                loadingState,
+                onLoadingFinished,
+                onLoadingStarted,
+                projectChanged,
+                requestProjectUpload: requestProjectUploadProp,
+                userOwnsProject,
+                /* eslint-enable no-unused-vars */
+                ...componentProps
+            } = this.props;
+            return (
+                <React.Fragment>
+                    <WrappedComponent
+                        onStartSelectingFileUpload={this.handleStartSelectingFileUpload}
+                        {...componentProps}
+                    />
+                    {fileInput}
```

Comment:
Let's try to create this element dynamically in the handler, rather than mounting it here, to avoid issues with components unmounting.